blog/2024-06-14-using-table-component.md

```diff
@@ -0,0 +1,486 @@
+
+
+
+```python
+class State(rx.State):
+    """The app state."""
+
+    users: list[Customer] = []
+    current_user: Customer = Customer()
+
+    ...
+
+    def get_user(self, user: Customer):
+        self.current_user = user
+
+    def update_customer_to_db(self, form_data: dict):
+        self.current_user.update(form_data)
+        with rx.session() as session:
+            customer = session.exec(
+                select(Customer).where(Customer.id == self.current_user["id"])
+            ).first()
+            for field in Customer.get_fields():
+                if field != "id":
+                    setattr(customer, field, self.current_user[field])
+            session.add(customer)
+            session.commit()
+        self.load_entries()
+        return rx.window_alert(f"User \{self.current_user["name"]} has been modified.")
+
+    def delete_customer(self, id: int):
+        """Delete a customer from the database."""
+        with rx.session() as session:
+            customer = session.exec(select(Customer).where(Customer.id == id)).first()
+            session.delete(customer)
+            session.commit()
+        self.load_entries()
+        return rx.window_alert(f"User \{self.current_user["name"]} has been deleted.")
+        
+
+def update_customer_dialog(user):
+    return rx.dialog.root(
+        rx.dialog.trigger(
+            rx.button(
+                rx.text("Edit", size="3"),
+                on_click=lambda: State.get_user(user),
+            ),
+        ),
+
+        ...
+
+            # Email
+            form_field(
+                "Email",
+                "user@reflex.dev",
+                "email",
+                "email",
+                user.email,
+            ),
+        
+        ...
+
+        on_submit=State.update_customer_to_db,
+        reset_on_submit=False,
+    )
+
+    ...
+
+def show_customer(user: Customer):
+    """Show a customer in a table row."""
+
+    return rx.table.row(
+        rx.table.cell(user.name),
+        rx.table.cell(user.email),
+        rx.table.cell(user.phone),
+        rx.table.cell(user.address),
+        rx.table.cell(user.payments),
+        rx.table.cell(user.date),
+        rx.table.cell(rx.match(
+            user.status,
+            ("Delivered", status_badge("Delivered")),
+            ("Pending", status_badge("Pending")),
+            ("Cancelled", status_badge("Cancelled")),
+            status_badge("Pending")
+        )),
+        rx.table.cell(
+            rx.hstack(
+                update_customer_dialog(user),
+                rx.icon_button(
+                    rx.icon("trash-2"),
+                    on_click=lambda: State.delete_customer(getattr(user, "id")),
```

Comment:
```suggestion
                    on_click=lambda: State.delete_customer(user.id),
```
did this not work for some reason? should be equivalent and a bit nicer to read

Reply:
fixed